proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,176 @@
+
+To fix the case stated above as well as similar cases originating from the same cause, the validation logic inside of SDL core should be updated to cutoff / ignore unknown enums present in RPC requests.  These failure cases should instead be replaced with SUCCESS cases with warnings result codes.  
+To ensure backward compatibility with earlier versions of SDL core, mobile libraries should be updated to process, filter and format enums inside of RPC requests when a difference in RAPI sets have been detected.
+
+  ## Proposed solution
+
+  ### SDL Core changes
+
+  When SDL receives a request from Mobile Application it starts validation of all incoming parameters.
+  If a parameter includes unknown enum value, SDL core has to cut off such value from the parameter.
+  This means the following cases are possible:
+  1. Type of parameter is non-array enum (see example 1)
+      SDL core has to remove this parameter from the request since the value after cutting off becomes empty.
+      
+  2. Parameter is an array of enum types (see example 2)
+      SDL core has to remove unknown value from the array.
+      A case is possible when all values are unknown and thus get removed.
+      In this case, SDL core has to proceed the same way as in case 1.
+      
+  3. Parameter is part of the structure (see example 3)
+      SDL has to proceed the same way as in case #1 or #2 and remove this parameter from the structure.
+      However, if the parameter is mandatory the structure becomes invalid.
+      In this case, SDL core has to remove the whole structure from the request.
+      During this process, SDL has to proceed recursively from the very bottom level up to the top.
+
+  4. Parameter is a part of the structure which is a part of an array (see example 4)
+      SDL core has to process it the same way as in case #3 and remove the structure as an item of the array.
+  Once all the parameters are processed SDL core has to proceed with the request as usual.
+  If at least one value of at least one parameter was cut off, SDL core has to update the response to Mobile Application as follows:
```

Comment:
```suggestion
      SDL Core has to process it the same way as in case #3 and remove the structure as an item of the array.
  Once all the parameters are processed SDL Core has to proceed with the request as usual.
  If at least one value of at least one parameter was cut off, SDL Core has to update the response to mobile application as follows:
```

Also, please see comment above regarding line breaks.